Sources/_StringProcessing/MatchingOptionSet.swift

```diff
@@ -0,0 +1,114 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import _MatchingEngine
+
+/// A set of matching options.
+struct MatchingOptionSet: RawRepresentable {
+  var rawValue: UInt32
+}
+
+extension MatchingOptionSet {
+  init(_ kind: AST.MatchingOption.Kind) {
+    self.rawValue = 1 << kind.rawValue
```

Comment:
Hmm.... are we certain that syntactic options correspond 1-1 to semantic concerns?